src/sqlfluff/core/parser/grammar/delimited.py

```diff
@@ -1,18 +1,18 @@
 """Definitions for Grammar."""
 
-from typing import Optional, Tuple, List
+from typing import Tuple, List
 
 from ..segments import BaseSegment
 from ..helpers import trim_non_code
 from ..match_result import MatchResult
 from ..match_wrapper import match_wrapper
 from ..context import ParseContext
 
-from .base import BaseGrammar
 from .noncode import NonCodeMatcher
+from .anyof import OneOf
 
 
-class Delimited(BaseGrammar):
+class Delimited(OneOf):
```

Comment:
Nice find here 🔍

Reply:
Yeah - originally these were implemented completely separately, but as the code matured, they ended up doing very similar things, so it wasn't too hard to just use some of the same implementation. 👍